### PR TITLE
Implement url analyzer API for import onboarding

### PR DESCRIPTION
**Pull request by jeroenpf**
#### Changes proposed in this Pull Request

This change implements the API endpoint for the URL analyzer in the import onboarding flow at `/start/import/capture`

#### Testing instructions

 - Go to `/start/import/capture`.
 - Enter a valid URL and hit enter
 - The scanning step will show - possibly really quick - followed by the preview page.
 - Check if the site address and the platform are correctly displayed in the content below the page title.

Not all functionality has been fully implemented yet. This PR purely focuses on getting the data from the URL analyzer endpoint and making it available in the step components.

Related to #57131

### CODE REVIEW

File: client/lib/wpcom-undocumented/lib/undocumented.js
Comment (pento) [inline client/lib/wpcom-undocumented/lib/undocumented.js]: Perhaps call this function `ImportsAnalyzeUrl()`? It's usually a good idea to match the function name with the API call.

It needs a docblock, too. 🙂

**Comment by pento** [comment]
This looks like it's heading in the right direction, nice work on getting Redux changes happening! There are a few bugs I noticed while I was testing, in no particular order:

- If you put in a fake URL, it correctly throws you back to the capture page, but doesn't give you an error message to explain what happened.
- On the preview page, there are a few text errors:
  - The URL has a trailing `/`.
  - The platform name needs to use the correct case (eg, WordPress, not wordpress).
  - "...is hosted by wordpress" sounds kind of odd for a self-hosted site. Perhaps we could use different wording here?
- There are a couple of special cases where we probably need to give an entirely different response:
  - When we don't know what the platform is.
  - When the site is already on WordPress.com.
- The preview image shows a loading indicator whilst mShots is generating a preview, but then doesn't load the proper preview.

Not all of these bugs need to be addressed in this PR, but let's make sure we keep track of them if they aren't.

File: client/signup/steps/import/capture/index.tsx
Comment (vishnugopal) [inline client/signup/steps/import/capture/index.tsx]: Remove this comment?

File: client/signup/steps/import/index.tsx
Comment (vishnugopal) [inline client/signup/steps/import/index.tsx:22]: Remove this?

**Review by vishnugopal — COMMENTED** [review]
A few bits of unneeded code, but otherwise looks good!

File: client/signup/steps/import/capture/index.tsx
Comment (jeroenpf) [reply client/signup/steps/import/capture/index.tsx]: @vishnugopal I left it there because the logic here is not complete and Bogdan has added the comment for when this will be finished. I will leave it here for now.

**Comment by jeroenpf** [comment]
> * If you put in a fake URL, it correctly throws you back to the capture page, but doesn't give you an error message to explain what happened.

I added some basic error handling. It will display the error that the URL is invalid.

> * On the preview page, there are a few text errors:
>   
>   * The URL has a trailing `/`.
>   * The platform name needs to use the correct case (eg, WordPress, not wordpress).
>   * "...is hosted by wordpress" sounds kind of odd for a self-hosted site. Perhaps we could use different wording here?
> * There are a couple of special cases where we probably need to give an entirely different response:
>   
>   * When we don't know what the platform is.
>   * When the site is already on WordPress.com.
> * The preview image shows a loading indicator whilst mShots is generating a preview, but then doesn't load the proper preview.

I will take note of these issues so we can take care of them in the next sprint.

**Comment by vishnugopal** [comment]
@jeroenpf Can we whip up the changes required to address feedback from Gary as well now? Thanks!

**Comment by vishnugopal** [comment]
I'm also getting this rather strange error when I try to test:

https://user-images.githubusercontent.com/533/139787210-5e2a2952-c851-4b2d-8e9d-6ef16185577a.mp4

**Comment by jeroenpf** [comment]
> I'm also getting this rather strange error when I try to test:
> 
>  CleanShot.2021-11-02.at.10.01.00.mp4

Looking into this now. I am not sure what is causing this.

**Comment by jeroenpf** [comment]
@vishnugopal I've seen the same issue now and it seems to be related to the dependency that was not used. I removed it now: 5cccaf4119ddd314c6aaa01d24c95a6aa534ddc8

**Comment by jeroenpf** [comment]
@vishnugopal @pento Most of the issues that @pento pointed out have been addressed in this PR.

Bogdan will give it a finishing touch and add a loading spinner for the preview image.  I've only implemented the logic that ensures the preview is shown correctly once it's available but wasn't sure about styling the loading state.

After that, two items are remaining:

- What to show when the site is hosted on WordPress.com - I guess we need to change the content so that it shows instructions for site migration instead? - In any case, this will also require a change in the API endpoint as it does not distinguish between wpcom and self-hosted WordPress sites.
- The language 'is hosted by [platform]' - What should we change this into? Something that works for any platform would be best. Perhaps: ' It looks like [website] is built with [platform] .... `


I've requested a review again from both of you, I'll let you guys decide who gets the honor to do it :-)

**Comment by vishnugopal** [comment]
Haven't re-reviewed, but:

> What to show when the site is hosted on WordPress.com - I guess we need to change the content so that it shows instructions for site migration instead? - In any case, this will also require a change in the API endpoint as it does not distinguish between wpcom and self-hosted WordPress sites. 

Am ok with this moving to a future PR.

> The language 'is hosted by [platform]' - What should we change this into? Something that works for any platform would be best. Perhaps: ' It looks like [website] is built with [platform] .... `

cc @ollierozdarz

**Comment by ollierozdarz** [comment]
👋  I think this answers @jeroenpf's questions above. 

Screen Shot 2021-11-03 at 2 35 00 pm

Also, here's the [Figma](https://www.figma.com/file/TAK5FpwTvfbENJDfHc7Aq1/Site-Importing?node-id=1930%3A21705) file.

**Comment by vishnugopal** [comment]
> The language 'is hosted by [platform]' - What should we change this into? Something that works for any platform would be best. Perhaps: ' It looks like [website] is built with [platform] .... `

"built with [platform]" sounds great here @jeroenpf and @ollierozdarz concurs!

**Comment by bogiii** [comment]
I've stumbled upon a warning when I load the capture page. We should address it before merging.
```
react-jsx-runtime.development.js:117 Warning: Failed prop type: The prop `stepName` is marked as required in `NavigationLink`, but its value is `undefined`.
    at NavigationLink (app:///./signup/navigation-link/index.jsx:43:5)
    at _class (app:///../packages/i18n-calypso/src/localize.js:36:9)
    at ConnectFunction (app:///../node_modules/react-redux/es/components/connectAdvanced.js:231:68)
    at div
    at ActionButtons (app:///../packages/onboarding/src/action-buttons/index.tsx:29:3)
    at div
    at StepWrapper (app:///./signup/step-wrapper/index.jsx:32:1)
    at _class (app:///../packages/i18n-calypso/src/localize.js:36:9)
    at ImportOnboarding (app:///./signup/steps/import/index.tsx:32:3)
    at ConnectFunction (app:///../node_modules/react-redux/es/components/connectAdvanced.js:231:68)
    at div
    at div
    at div
    at div
    at Signup (app:///./signup/main.jsx:151:5)
    at ConnectFunction (app:///../node_modules/react-redux/es/components/connectAdvanced.js:231:68)
    at div
    at div
    at div
    at Layout (app:///./layout/index.jsx:243:1)
    at ConnectFunction (app:///../node_modules/react-redux/es/components/connectAdvanced.js:231:68)
    at WithCurrentRoute (app:///./components/route/index.js:44:30)
    at MomentProvider (app:///./components/localized-moment/provider.js:27:5)
    at ConnectFunction (app:///../node_modules/react-redux/es/components/connectAdvanced.js:231:68)
    at Provider (app:///../node_modules/react-redux/es/components/Provider.js:18:20)
    at QueryClientProvider (app:///../node_modules/react-query/es/react/QueryClientProvider.js:38:21)
    at RouteProvider (app:///./components/route/index.js:22:3)
    at I18nProvider (app:///../node_modules/@wordpress/react-i18n/build-module/index.js:68:5)
    at LocaleProvider (app:///../packages/i18n-utils/src/locale-context.tsx:19:3)
    at CalypsoI18nProvider (app:///./components/calypso-i18n-provider/index.tsx:20:3)
    at ProviderWrappedLayout (app:///./controller/index.web.js:60:3)
```

**Review by bogiii — APPROVED** [review]
We are good to go! 🚀

**Comment by tyxla** [comment]
Hey, @jeroenpf @bogiii @pento @vishnugopal! 

We're in the process of migrating away from the deprecated `wpcom.undocumented`, and I just noticed that this PR adds a new method to it. Would you mind updating to use `wpcom.req.get()` or `wpcom.req.post()` instead? 

If you take a look at [the history](https://github.com/Automattic/wp-calypso/commits/trunk/client/lib/wpcom-undocumented/lib/undocumented.js), there's a bunch of recent refactors that can be used as a reference. Also see p4TIVU-9KX-p2 for more details, and feel free to ping me or anyone from @Automattic/team-calypso-frameworks if you need any help.

Thanks in advance!

File: client/lib/wpcom-undocumented/lib/undocumented.js
Comment (jsnajdr) [inline client/lib/wpcom-undocumented/lib/undocumented.js:1664]: By the way the `method` field doesn't need to be there, calling `req.get` already says what HTTP method should be used.

Comment (jeroenpf) [reply client/lib/wpcom-undocumented/lib/undocumented.js:1664]: @jsnajdr Thanks for pointing this out :-) I've already created a new PR to address your feedback: #57815 - It will be merged by tomorrow.

**Pull request merged**
